crates/solver/src/settlement/settlement_encoder.rs

```diff
@@ -555,6 +557,17 @@ impl SettlementEncoder {
     }
 }
 
+fn verify_executed_amount(order: &Order, executed_amount: U256) -> Result<()> {
+    let valid_executed_amount = match (order.creation.partially_fillable, order.creation.kind) {
+        (true, OrderKind::Sell) => executed_amount <= order.creation.sell_amount,
+        (true, OrderKind::Buy) => executed_amount <= order.creation.buy_amount,
```

Comment:
You're right. I was thinking this would be detected during simulation because the contract enforces it too but it probably makes sense to catch it as early as possible.

Reply:
This was a bit annoying because the other number is a BigInt so would have to convert and then decide what to do if that value doesn't make sense like saturating sub so I left it out for now. Will think again how I want to do this.